### PR TITLE
Add StatefulSets checks at Service level

### PR DESCRIPTION
**Pull request by slaskawi**
Hi!

Please let me propose some very small e2e testsuite enhancement. 

This PR removed a `TODO` about checking governing service at unit test level (which is hard) and adds this to e2e testsuite.

Thanks
Sebastian

### CODE REVIEW

**Comment by slaskawi** [comment]
I signed it!

File: test/e2e/petset.go [hunk 1]
```diff
@@ -963,6 +966,16 @@ func (p *statefulSetTester) waitForStatus(ps *apps.StatefulSet, expectedReplicas
 	}
 }
 
+func (p *statefulSetTester) checkServiceName(ps *apps.StatefulSet, expectedServiceName string) error {
+	framework.Logf("Checking if statefulset spec.serviceName is %s", expectedServiceName)
+
+	if expectedServiceName != ps.Spec.ServiceName {
+		return fmt.Errorf("Wrong service name gouverning statefulset. Expected %s got %s", expectedServiceName, ps.Spec.ServiceName)
```
Comment (foxish) [inline test/e2e/petset.go]: nit: gouverning -> governing

**Review by foxish — COMMENTED** [review]
LGTM except one nit

File: test/e2e/petset.go (hunk 1 above)
Comment (slaskawi) [inline test/e2e/petset.go]: ooh sorry, I need to fix my spell checker :)

**Comment by slaskawi** [comment]
Rebased

**Comment by foxish** [comment]
@fejta PTAL, waiting for your approval on the test directory changes.

**Pull request merged**